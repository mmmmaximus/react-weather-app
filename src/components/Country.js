import React from 'react'
import Weather from './Weather'

const Country = ({country}) => {
  return (
    <div>
      <h2>{country.name}</h2>
      <p>capital {country.capital}</p>
      <p>population {country.population}</p>
      <h3>languages</h3>
      {country.languages.map((language, index) => 
        <li key={index}>{language.name}</li>
      )}
      <img src={country.flag} width="150" height="150" alt={country.name}/>
      <h3>Weather in {country.name}</h3>
      <Weather city={country.capital}/>
    </div>
  )
}

export default Country
