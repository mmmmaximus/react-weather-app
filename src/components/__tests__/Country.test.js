import { render, screen } from '@testing-library/react'
import Country from '../Country'
import renderer from 'react-test-renderer'
import mockedAxios from '../../__mocks__/axios'

describe('Country component', () => {
  it('should render component', () => {
    const weatherResponse = {
      data: {
        data: [null]
      }
    }
    mockedAxios.get.mockResolvedValueOnce(weatherResponse)

    const country = {
      name: 'Africa',
      capital: 'Wakanda',
      population: '123456',
      languages: [
        {name: 'language1'},
        {name: 'language2'},
        {name: 'language3'}
      ]
    }
    render(<Country country={country}/>)

    // renders component
    const CountryElement =  screen.getByTestId('Country')
    expect(CountryElement).toBeInTheDocument

    // renders content in component
    expect(CountryElement.textContent).toContain('Africa')
    expect(CountryElement.textContent).toContain('Wakanda')
    expect(CountryElement.textContent).toContain('123456')
    expect(CountryElement.textContent).toContain('language1')
    expect(CountryElement.textContent).toContain('language2')
    expect(CountryElement.textContent).toContain('language3')
    expect(screen.getByTestId('img')).toBeInTheDocument
  })

  it('matches snapshot', () => {
    const country = {
      name: 'Africa',
      capital: 'Wakanda',
      population: '123456',
      languages: [
        {name: 'language1'},
        {name: 'language2'},
        {name: 'language3'}
      ]
    }
    const tree = renderer.create(<Country country={country} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
