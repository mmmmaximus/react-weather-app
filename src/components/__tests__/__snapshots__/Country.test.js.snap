// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country component matches snapshot 1`] = `
<div
  data-testid="Country"
>
  <h2>
    Africa
  </h2>
  <p>
    capital 
    Wakanda
  </p>
  <p>
    population 
    123456
  </p>
  <h3>
    languages
  </h3>
  <li>
    language1
  </li>
  <li>
    language2
  </li>
  <li>
    language3
  </li>
  <img
    alt="Africa"
    data-testid="img"
    height="150"
    width="150"
  />
  <h3>
    Weather in 
    Africa
  </h3>
  <div />
</div>
`;
